File: backend/internal/domain/photo.go
package domain

import "time"

// Photo は画像投稿のメイン情報
type Photo struct {
	ID          int64
	ImageURL    string
	AspectRatio float64
	Title       *string
	Description *string
	CategoryID  *int64
	UserID      *int64
	IsVisible   bool
	TakenAt     *time.Time
	Tags        []string
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

// 写真詳細
type PublicPhotoDetail struct {
	Photo *Photo
	Exif  *PhotoExif
	GPS   *PhotoGPS
	Tags  []*Tag
}

File: frontend/src/types/public/photo.ts
import { ExifInfo } from "../photoExif";
import { GPSInfo } from "../photoGps";
import { Tag } from "../tag";

// 基本の型
export type PublicPhoto = {
  id: number;
  url: string;
  aspectRatio: number;
  title?: string | null;
  description?: string | null;
  takenAt?: string | null;
  sortOrder: number;
};

// 詳細画面用
export type PublicPhotoDetail = {
  id: number;
  imageUrl: string;
  aspectRatio: number;
  title?: string | null;
  description?: string | null;
  takenAt?: string | null;
  exif?: ExifInfo;
  gps?: GPSInfo;
  tags: Tag[];
};
